crates/runtime/benches/vector_search.rs

```diff
@@ -198,23 +218,52 @@ async fn run_search_queries(
                     chunk_latency.as_millis()
                 );
 
-                Ok(response_time)
+                Ok((response_time, scores))
             }
         })
         .buffer_unordered(num_chunks_in_parallel)
-        .try_collect::<Vec<Vec<f64>>>()
+        .try_collect::<Vec<(Vec<f64>, Vec<(String, HashMap<String, f64>)>)>>()
         .await
         .inspect_err(|_| {
             benchmark_result.finish_search();
         })?
         .into_iter()
-        .for_each(|response_time| {
+        .for_each(|(response_time, scores)| {
             for time in response_time {
                 benchmark_result.record_response_time(time);
             }
+
+            result.extend(scores);
         });
 
     benchmark_result.finish_search();
 
-    Ok(())
+    Ok(result)
+}
+
+fn to_search_result(result: &VectorSearchResult) -> Result<HashMap<String, f64>, String> {
+    let mut output = HashMap::new();
+
+    for (table_ref, value) in result {
+        match value.to_matches(table_ref) {
+            Ok(matches) => {
+                for m in matches {
+                    let id = m.metadata().get("_id").ok_or_else(|| {
```

Comment:
If you set `id` as primary key you can then use `VectorSearchTableResult::primary_keys` (to manually set these, you'd just have to set `explicit_primary_keys` in `VectorSearch::new(...`).

Reply:
Thank you, the `to_matches` is included to benchmark its usage as we use it in `v1/search`, so the benchmark logic is closer to the target api usage